### DISCUSSION TITLE
Informix (JDBC, ODBC)

### DISCUSSION THREAD

**Discussion by swright-eti** [directus/directus#15933] [Archive]
Does Directus support Informix as a DB_CLIENT?

**Comment by azrikahar** [comment]
It is not supported as of now. 

Technically Knex is used under the hood, so based on what I was able to briefly find over here: https://github.com/knex/knex/issues/1350, it _might_ be possible to fork Directus and use https://github.com/henryjw/knex-db2 as an additional Knex dialect. However do note that it may be non-trivial to implement, as there seems to be some limitations such as no transaction support (which is used in Directus services for example).

It seems like there was also plan to use [`ibm_db`](https://github.com/ibmdb/node-ibm_db) as stated here: https://github.com/henryjw/knex-db2/issues/12, but it's not actively worked on.

**Reply by swright-eti** [reply]
Thanks, I appreciate the information.

**Reply by azrikahar** [reply]
Out of curiosity as I'm not familiar with it but it does look interesting, is there a specific use case for this? Perhaps it's an existing database that you would like to try out Directus with?

**Reply by swright-eti** [reply]
Its the database that the company I work for started using back around 2000. At the time it was a reasonably good choice for the feature set and licensing costs. But unfortunately its actually become a bit of a burden. The licensing has become very expensive and the features we need are now available in things like MariaDB. So I would not suggest starting any new projects using it.

**Reply by azrikahar** [reply]
Thanks for sharing! That does seem unfortunate 🙁 Here's to hoping that you'll still be able to give Directus a spin on one of the currently supported databases in other/future projects 👍
